sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -544,6 +551,23 @@ export class StudentService extends RecordDataModelService<Student> {
     if (noteType) {
       studentNoteQuery.andWhere("note.noteType = :noteType", { noteType });
     }
+    if (options?.filterNoEffectRestrictionNotes) {
+      const filterSubQuery = this.dataSource
+        .getRepository(StudentRestriction)
+        .createQueryBuilder("studentRestriction")
+        .select("note.id")
+        .innerJoin("studentRestriction.restrictionNote", "note")
+        .innerJoin("studentRestriction.restriction", "restriction")
+        .innerJoin("studentRestriction.student", "student")
+        .where("student.id = :studentId", { studentId })
+        .andWhere(
+          "restriction.notificationType = :restrictionNotificationType",
+          { restrictionNotificationType: RestrictionNotificationType.NoEffect },
+        );
+      studentNoteQuery
+        .andWhere("note.id not in  (" + filterSubQuery.getQuery() + ")")
```

Comment:
There is at least one extra white space in the string.
As a recommendation, I would prefer string interpolation for this case:
`note.id not in (${filterSubQuery.getQuery()})`